### PR TITLE
Fix redirect empty backend

### LINKED ISSUE #679: Entrypoint redirect to https rule breaks frontend rules
I created a docker-compose project to recreate this issue
https://github.com/iadknet/traefik-ssl-redirect

```
docker-compose up
```

Make an http request to the /test_app and get redirected to https (the redirect works fine)

```
$ curl -L -k -v http://localhost/test_app
*   Trying ::1...
* connect to ::1 port 80 failed: Connection refused
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 80 (#0)
> GET /test_app HTTP/1.1
> Host: localhost
> User-Agent: curl/7.43.0
> Accept: */*
>
< HTTP/1.1 302 Found
< Server: nginx/1.11.4
< Date: Thu, 15 Sep 2016 23:08:10 GMT
< Content-Type: text/plain; charset=utf-8
< Content-Length: 5
< Connection: keep-alive
< Location: https://localhost/test_app
<
* Ignoring the response-body
* Connection #0 to host localhost left intact
* Issue another request to this URL: 'https://localhost/test_app'
* Found bundle for host localhost: 0x7f8d03e00790
*   Trying ::1...
* connect to ::1 port 443 failed: Connection refused
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 443 (#1)
* TLS 1.2 connection using TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
* Server certificate: dev.evisions
> GET /test_app HTTP/1.1
> Host: localhost
> User-Agent: curl/7.43.0
> Accept: */*
>
< HTTP/1.1 200 OK
< Server: nginx/1.11.4
< Date: Thu, 15 Sep 2016 23:08:10 GMT
< Content-Type: text/plain; charset=utf-8
< Content-Length: 0
< Connection: keep-alive
<
* Connection #1 to host localhost left intact
```

The final request returns a 200, but the content is empty.

nginx logs:

```
nginx_1    | 172.19.0.1 - - [15/Sep/2016:23:08:10 +0000] "GET /test_app HTTP/1.1" 302 5 "-" "curl/7.43.0" "-"
nginx_1    | 172.19.0.1 - - [15/Sep/2016:23:08:10 +0000] "GET /test_app HTTP/1.1" 200 0 "-" "curl/7.43.0" "-"
```

traefik logs:

```
172.19.0.2 - - [15/Sep/2016:23:08:10 +0000] "GET /test_app HTTP/1.0" 0 0 "" "curl/7.43.0" 6 "" "" 90.389µs
```

To test without the redirect rules in place:

Open `traefik/traefik.toml` and comment out the following lines:

```
     [entryPoints.http.redirect]
       regex = "^http://(.*)"
       replacement = "https://$1"
```

```
docker-compose down && docker-compose up
```

Curl https://localhost/ and get the full page:

```
$ curl -L -k -v https://localhost/test_app
*   Trying ::1...
* connect to ::1 port 443 failed: Connection refused
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 443 (#0)
* TLS 1.2 connection using TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
* Server certificate: dev.evisions
> GET /test_app HTTP/1.1
> Host: localhost
> User-Agent: curl/7.43.0
> Accept: */*
>
< HTTP/1.1 200 OK
< Server: nginx/1.11.4
< Date: Thu, 15 Sep 2016 23:10:56 GMT
< Content-Type: text/html; charset=utf-8
< Content-Length: 118
< Connection: keep-alive
< Etag: W/"76-ktOkqas+2XplZPRxS5r6LA"
< Vary: Accept
< X-Powered-By: Express
<
<!DOCTYPE html>
<html>
<head>


  <title>Test App</title>

</head>
<body>
  <h1>Test Application</h1>
</body>
</html>
* Connection #0 to host localhost left intact
```

nginx logs:

```
nginx_1    | 172.19.0.1 - - [15/Sep/2016:23:12:10 +0000] "GET /test_app HTTP/1.1" 200 118 "-" "curl/7.43.0" "-"
```

traefik logs:

```
172.19.0.3 - - [15/Sep/2016:23:12:10 +0000] "GET / HTTP/1.0" 200 118 "" "curl/7.43.0" 2 "web-test-service" "http://172.19.0.5:8080" 1.487498ms
```

### PR DESCRIPTION
**Pull request by argeiger**
Fixes #679

PTAL @emilevauge 

One thing I was considering is verifying backends[frontend.Backend] wasn't null around line 756.  Let me know if you have any questions.

### CODE REVIEW

**Comment by timoreimann** [comment]
That's quite a diff I must say. I'm a bit hesitant to approve this as I'm uncertain how it may affect the overall behavior, especially with regards to undesirable side effects.

I'm thinking we should take the opportunity and start to cover the changed functionality by tests in order to bring some safety.

**Comment by argeiger** [comment]
@timoreimann, the change very little (5-6 things).  What the diff shows is the fix for the indentation after I took the else statement out.  What the change does is add the redirect handler on the defined routes so the other route handler will send it to the proper backend and not get an empty body back.  Let me know if you have any questions.

File: server.go [hunk 1]
```diff
@@ -598,99 +598,66 @@ func (server *Server) loadConfig(configurations configs, globalConfiguration Glo
 					log.Debugf("Creating route %s %s", routeName, route.Rule)
 				}
 				entryPoint := globalConfiguration.EntryPoints[entryPointName]
+				var negroni = negroni.New()
```
Comment (timoreimann) [inline server.go]: Can be shortened to `negroni := negroni.New()`.

File: server.go [hunk 2]
```diff
@@ -598,99 +598,66 @@ func (server *Server) loadConfig(configurations configs, globalConfiguration Glo
...
 					} else if handler, err := server.loadEntryPointConfig(entryPointName, entryPoint); err != nil {
 						log.Errorf("Error loading entrypoint configuration for frontend %s: %v", frontendName, err)
 						log.Errorf("Skipping frontend %s...", frontendName)
 						continue frontend
 					} else {
-						newServerRoute.route.Handler(handler)
+						negroni.Use(handler)
 						redirectHandlers[entryPointName] = handler
 					}
-				} else {
```
Comment (timoreimann) [inline server.go]: I wonder why there was an `else` here in the first place. Was this an oversight or a true requirement?

Any thoughts?

**Comment by timoreimann** [comment]
@emilevauge should we merge this into v1.2 instead of master? #679 sounds fairly important to me.

**Comment by argeiger** [comment]
@timoreimann, thanks for taking the time to review and the suggestion.  I added it/merged/committed so all should be there now.  Thanks again.

**Comment by emilevauge** [comment]
Thanks @argeiger, I carried your PR in #1417 :)

**Pull request closed without merging**